pkg/metrics/producers/scheduledcapacity/producer.go

```diff
@@ -25,5 +28,56 @@ type Producer struct {
 
 // Reconcile of the metrics
 func (p *Producer) Reconcile() error {
+	var (
+		loc     *time.Location
+		err     error
+		currVal int32
+	)
+	matchFound := false
+	now := time.Now()
+
+	if p.Spec.Schedule.Timezone != nil {
+		loc, err = time.LoadLocation(*p.Spec.Schedule.Timezone)
+		if err != nil {
+			return fmt.Errorf("timezone was not a valid input")
+		}
+	} else {
+		// Set Default location to UTC if not specified
+		loc = time.UTC
+	}
+	now = now.In(loc)
+
+	for _, behavior := range p.Spec.Schedule.Behaviors {
+		// use Cron library to find the next time start and end next match
+		startTime, err := nextTime(crontabFrom(behavior.Start), loc)
+		if err != nil {
+			return fmt.Errorf("start pattern is invalid: %w", err)
+		}
+		endTime, err := nextTime(crontabFrom(behavior.End), loc)
+		if err != nil {
+			return fmt.Errorf("end pattern is invalid: %w", err)
+		}
+
+		if !now.After(endTime) && (!endTime.After(startTime) || !startTime.After(now)) {
+			// Since the way collisions are handled are by how they're ordered in the spec, stop on first match
+			currVal = behavior.Replicas
+			matchFound = true
+		}
+	}
+	if !matchFound {
```

Comment:
Can you skip this boolean and just nil check?
```
p.Status.Schedule.CurrentValue = p.Spec.Schedule.DefaultReplicas
// Loop and try to find a non default
for ... {
  if predicate {
    p.Status.Schedule.CurrentValue = behavior.Replicas
  } 
}
```

Reply:
Ah yeah you make a good point. Will change it.

Reply:
I added the boolean check back here because we don't know if the `scheduledCapacity` will be nil necessarily or not. With that, it would be two nil checks. One for `scheduledCapacity` and one for `scheduledCapacity.CurrentValue`. I think this is cleaner.

Reply:
```
// schedule can't be nil
p.Status.Schedule = &v1alpha1.ScheduledCapacityStatus{
  CurrentValue: p.Spec.Schedule.DefaultReplicas, //p.spec.Scheudule can't be nil since we're in this producer
} 
```